x/auth/tx/sigs.go

```diff
@@ -0,0 +1,104 @@
+package tx
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+)
+
+// SignatureDataToModeInfoAndSig converts a SignatureData to a ModeInfo and raw bytes signature
+func SignatureDataToModeInfoAndSig(data signing.SignatureData) (*tx.ModeInfo, []byte) {
+	if data == nil {
+		return nil, nil
+	}
+
+	switch data := data.(type) {
+	case *signing.SingleSignatureData:
+		return &tx.ModeInfo{
+			Sum: &tx.ModeInfo_Single_{
+				Single: &tx.ModeInfo_Single{Mode: data.SignMode},
+			},
+		}, data.Signature
+	case *signing.MultiSignatureData:
+		n := len(data.Signatures)
+		modeInfos := make([]*tx.ModeInfo, n)
+		sigs := make([][]byte, n)
+
+		for i, d := range data.Signatures {
+			modeInfos[i], sigs[i] = SignatureDataToModeInfoAndSig(d)
+		}
+
+		multisig := types.MultiSignature{
+			Signatures: sigs,
+		}
+		sig, err := multisig.Marshal()
+		if err != nil {
+			panic(err)
+		}
+
+		return &tx.ModeInfo{
+			Sum: &tx.ModeInfo_Multi_{
+				Multi: &tx.ModeInfo_Multi{
+					Bitarray:  data.BitArray,
+					ModeInfos: modeInfos,
+				},
+			},
+		}, sig
+	default:
+		panic("unexpected case")
```

Comment:
can we wrap the panic with the data error? 
```suggestion
		panic(fmt.Sprintf("unexpected signature data type %T", data))
```